cli/azd/pkg/input/previewer.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package input
+
+import (
+	"strings"
+
+	tm "github.com/buger/goterm"
+)
+
+type previewer struct {
+	title  string
+	lines  int
+	prefix string
+	output []string
+}
+
+func NewPreviewer(lines int, prefix string) *previewer {
+	return &previewer{
+		lines:  lines,
+		output: make([]string, lines),
+		prefix: prefix,
+	}
+}
+
+func (p *previewer) SetTitle(title string) {
+	p.title = title
+}
+
+func (p *previewer) Start() {
+	printOutput(p.output, p.title)
+}
+
+func clearString(text string) {
+	sizeLog := len(text)
+	eraseWith := strings.Repeat(" ", sizeLog)
+	tm.MoveCursorUp(1)
+	tm.Printf(eraseWith)
+	tm.MoveCursorBackward(sizeLog)
+}
+
+func (p *previewer) Stop() {
+	for _, log := range p.output {
+		clearString(log)
+	}
+	if p.title != "" {
+		clearString(p.title)
+	}
+	tm.Flush()
+}
+
+func (p *previewer) Write(logBytes []byte) (int, error) {
+	lastChar := len(logBytes) - 1
+	if logBytes[lastChar] == '\n' {
+		logBytes = logBytes[:lastChar]
+	}
+	log := string(logBytes)
+	maxWidth := tm.Width()
+	fullLog := p.prefix + log
+	if len(fullLog) > maxWidth {
+		fullLog = fullLog[:maxWidth-4] + cPostfix
+	}
+
+	p.Stop()
+	p.output = pushRemove(p.output, fullLog)
+	printOutput(p.output, p.title)
+	return len(logBytes), nil
+}
+
+func pushRemove(original []string, value string) []string {
+	copy := original[1:] // remove first
+	return append(copy, value)
+}
+
+func printOutput(output []string, title string) {
+	size := len(output)
+	count := size
+	if title != "" {
+		tm.ResetLine("")
+		tm.Println(title)
+	}
+	for count > 0 {
+		count--
+		tm.ResetLine("")
+		tm.Println(output[count])
+	}
+	tm.Flush()
+}
```

Comment:
Have no strong feelings here.
I ended up using top-first as a way to give the previewer a way to be identified (a user can refer about this as the azd previewer way). This would help during issues/debug as users will know they are looking to either the classic azd output or the azd previewer.

But reverting to bottom-first is easy.

So, yes, whatever @savannahostrowski / @Austinauth want here is OK